bigframes/_config/__init__.py

```diff
@@ -29,44 +32,93 @@ class Options:
     """Global options affecting BigQuery DataFrames behavior."""
 
     def __init__(self):
+        self._local = threading.local()
+
+        # Initialize these in the property getters to make sure we do have a
+        # separate instance per thread.
+        self._local.bigquery_options = None
+        self._local.display_options = None
+        self._local.sampling_options = None
+        self._local.compute_options = None
+
+        # BigQuery options are special because they can only be set once per
+        # session, so we need an indicator as to whether we are using the
+        # thread-local session or the global session.
+        self._local.is_bigquery_thread_local = False
```

Comment:
If I'm reading this correctly, would checking for self._local.bigquery_options is None be able to replace this boolean state? It seems a little more DRY to do it that way, if so.

Reply:
I like it! Done in https://github.com/googleapis/python-bigquery-dataframes/pull/652/commits/a7f3c967e0e1be92189c1123d33651901f4561b9